### DISCUSSION TITLE
1. Save index to local files; 2. Abitility to reorder status_bar item

### DISCUSSION THREAD

**Discussion by qingshuizheng** [ahrm/sioyek#629] [Q&A] [answered]
Hi @ahrm ,

1. Do we currently have any settings to enable saving indexing results to local tmp files? I have a pdf with 3400 pages, which would take some time to re-index every time I reopen it. 
2. Is re-ordering status_bar items possible? Since the 2nd item (perhaps "chapter name"?) is too jumpy due to variance in length of names. Personally I would like to put it at the end of status_bar. A .gif is attached:
![Kapture 2023-02-08 at 07 43 57](https://user-images.githubusercontent.com/78839107/217533234-9ac42631-8586-447a-866c-c66df9e92f82.gif)

Zheng.

**Comment by ahrm** [comment, marked as answer]
1. No.
2. Added in e5f6b7ecc5ec3f3c8653868e3c7936f6c4f4315b. Now you can customize the statusbar format like this:
```
statusbar_format Page %{current_page} / %{num_pages}%{search_results}%{search_progress}%{link_status}%{waiting_for_symbol}%{indexing}%{preview_index}%{synctex}%{drag}%{presentation}%{visual_scroll}%{locked_scroll}%{highlight}%{closest_bookmark}%{close_portal}%{rect_select}%{custom_message}%{chapter_name}
```

**Reply by qingshuizheng** [reply]
Thanks @ahrm , Glad to have the new feature.

For the indexing temp file idea, yeah probably we don't need this. It takes only 30s or so for a pdf of 3000 pages. And usually I don't close and reopen it too often. Just a gut feeling of no need to reindex every time if the contents are the same and the index file size is small.

Will try the new build when the GitHub action is done. 

Again thanks for your time!

**Reply by qingshuizheng** [reply]
Tested. Perfect! Thanks.

**Reply by qingshuizheng** [reply]
There's a naming consistency issue(?) i commented here: https://github.com/ahrm/sioyek/commit/e5f6b7ecc5ec3f3c8653868e3c7936f6c4f4315b#r99965906

But you might have your own considerations. 😄

**Reply by ahrm** [reply]
It is changed in 3d719af3ce910e8ca2abe798eb9b6ae25d7e1c0a.
